Automattic/jetpack
# My Jetpack: Add filter for avoiding initalization

Introduces a method `should_initialize` on the initializer with `jetpack_my_jetpack_should_initialize` filter.

#### Changes proposed in this Pull Request:

* Introduces a method `should_initialize` on the initializer..
* Adds a WordPress filter `jetpack_my_jetpack_should_initialize` for avoiding initalization

For Rationale, see the request for comments on this P2 p1HpG7-eqp-p2

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* On a new site, or old one and disconnect Jetpack
* Verify that the My Jetpack menu is not present
* Connect site at a site level only (Via the Jetpack plugin). Disconnect
* Verify that the My Jetpack menu appears
* Connect a user
* Verify that the My Jetpack menu appears
* Unset the constant `JETPACK_ENABLE_MY_JETPACK` and confirm the menu disappears. Set it back.
* Add some snippet somewhere with `add_filter( 'jetpack_my_jetpack_should_initialize', '__return_false')` and confirm My Jetpack does not load.
